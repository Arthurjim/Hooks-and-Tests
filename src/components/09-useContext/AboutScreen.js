import React, { useContext } from 'react'
import { UserContext } from './UserContext'

const AboutScreen = () => {
    const {user, setUser} = useContext(UserContext)
    const handleClcik =()=>{
        setUser({})
    }
    return (
        <div>
            <h1>AboutScreen</h1>
            <hr/>
            <pre>
                {JSON.stringify(user, null,3)}
            </pre>
            <button className="bnt btn-secondary" onClick={handleClcik}>
                Logout
            </button>
        </div>
    )
}

export default AboutScreen