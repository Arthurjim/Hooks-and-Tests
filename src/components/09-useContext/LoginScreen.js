import React,{useContext} from 'react'
import { UserContext } from './UserContext'

const LoginScreen = () => {
    const {setUser} = useContext(UserContext)
    const sendUser =()=>{
        const user ={
            id:123,
            name:'Arturo',
            email:'arthurjim013@gmail.com'
        }
        setUser(user)
    }
    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
                <button className="btn btn-primary" onClick={sendUser}>
                    Ingresar
                </button>
        </div>
    )
}

export default LoginScreen