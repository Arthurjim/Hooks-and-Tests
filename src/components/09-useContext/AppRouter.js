
import React from "react";
import {
  BrowserRouter as Router,
  Routes ,
  Route,
  Navigate
} from "react-router-dom";
import AboutScreen from './AboutScreen'
import LoginScreen from './LoginScreen'
import HomeScreen from './HomeScreen'
import NavBar from "./components/NavBar";

const AppRouter = () => {
    return (
       <Router>
           <div>
               <NavBar/>
               <div className="container">

                    <Routes >
                        <Route exact path="/"  element={<HomeScreen />}/>
                        <Route exact path="/about" element={<AboutScreen />}/>
                        <Route exact path="/login" element={<LoginScreen />}/>
                        {/* <Route element={<HomeScreen />}/> */}
                        <Route  path="*" element={<Navigate replace to="/" />} />  
                        
                    </Routes >
               </div>
           </div>
       </Router>
    )
}

export default AppRouter
