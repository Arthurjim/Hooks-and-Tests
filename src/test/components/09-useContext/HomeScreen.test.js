import { mount, shallow } from "enzyme"
import HomeScreen from '../../../components/09-useContext/HomeScreen'
import { UserContext } from "../../../components/09-useContext/UserContext"



describe('Pruebas en HomeScreen', () => {
    const user = {name:'Arturo',email:'Artuhr@gmail.com'}
    const wrapper = mount(
    <UserContext.Provider value={{user}}>
        <HomeScreen />
    </UserContext.Provider>
    )
    test('Debe de mostrar correctamenet ', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
})
