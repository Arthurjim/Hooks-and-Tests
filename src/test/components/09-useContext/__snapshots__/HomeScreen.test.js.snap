// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomeScreen Debe de mostrar correctamenet  1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre
    className="container"
  >
    {
   "name": "Arturo",
   "email": "Artuhr@gmail.com"
}
  </pre>
</div>
`;
